datadog_lambda/wrapper.py

```diff
@@ -234,6 +263,26 @@ def _after(self, event, context):
                 flush_stats()
             if should_use_extension:
                 flush_extension()
+
+            if (
+                self.encode_authorizer_context
+                and self.response
+                and self.response.get("principalId")
+                and self.response.get("policyDocument")
+            ):
+                # the finish_time_ns should be set as the end of the inferred span if it exist
+                #  or the end of the current span
+                reference_span = self.inferred_span if self.inferred_span else self.span
+                finish_time_ns = (
+                    reference_span.start_ns + reference_span.duration_ns
```

Comment:
Maybe this could be another method? Although I realize it's pretty python-y already. Up to you.

Reply:
I tried to refactor then I realized that the `finish_time_ns` calculation requires that the `reference_span` is finished already. 
So I'll skip the refactoring to save the `reference_span already finished` check. I'll add a comment instead.

Reply:
On a second thought, I decided to add the `reference_span already finished` check.